ocpp/v16/datatypes.py

```diff
@@ -121,3 +122,45 @@ class MeterValue:
 
     timestamp: str
     sampled_value: List[SampledValue]
+
+
+# Security Extension
+
+
+@dataclass
+class CertificateHashData:
+    """
+    CertificateHashDataType is used by:
+    DeleteCertificate.req, GetInstalledCertificateIds.conf
+    """
+
+    hash_algorithm: HashAlgorithm
+    issuer_name_hash: str
+    issuer_key_hash: str
+    serial_number: str
+
+
+@dataclass
+class Firmware:
+    """
+    Represents a copy of the firmware that can be loaded/updated on the Charge Point.
+    FirmwareType is used by: SignedUpdateFirmware.req
+    """
+
+    location: str
+    retrieve_date_time: str
+    install_date_time: Optional[str] = None
+    signing_certificate: Optional[str] = None
```

Comment:
This field is required, according to the a required field. 
![2023-11-15-141445_1045x675_scrot](https://github.com/mobilityhouse/ocpp/assets/1565144/ee1f3f8b-5e0a-4d55-83cb-2d35d028c6c9)

Reply:
Fixed. Good catch.